src/ploigos_step_runner/step_implementers/automated_governance/rekor.py

```diff
@@ -0,0 +1,177 @@
+from ploigos_step_runner.utils.io import create_sh_redirect_to_multiple_streams_fn_callback
+from ploigos_step_runner.utils.io import TextIOIndenter
+from ploigos_step_runner.utils.dict import deep_merge
+from ploigos_step_runner.utils.file import base64_encode
+from ploigos_step_runner.utils.file import get_file_hash
+from ploigos_step_runner.utils.gpg import get_gpg_key
+
+REQUIRED_CONFIG_OR_PREVIOUS_STEP_RESULT_ARTIFACT_KEYS = [
+    'rekor-server',
+    'gpg-key',
+    'gpg-user'
+]
+
+
+class Rekor(StepImplementer):  # pylint: disable=too-few-public-methods
+    """`StepImplementer` for the `automated-governance` step using Rekor.
+    """
+
+    @staticmethod
+    def _required_config_or_result_keys():
+        """Getter for step configuration or previous step result artifacts that are required before
+        running this step.
+
+        See Also
+        --------
+        _validate_required_config_or_previous_step_result_artifact_keys
+
+        Returns
+        -------
+        array_list
+            Array of configuration keys or previous step result artifacts
+            that are required before running the step.
+        """
+        return REQUIRED_CONFIG_OR_PREVIOUS_STEP_RESULT_ARTIFACT_KEYS
+
+    def create_rekor_entry( self,
+        public_key_path,
+        gpg_user,
+        extra_data_file
+    ):
+        """Creates a rekor entry as a dict object.
+
+        Returns
+        -------
+        StepResult
+            Dict that contains rekor entry for upload with the cli
+        """
+        hash = get_file_hash(extra_data_file)
+        sig_file = extra_data_file + '.asc'
+        sig_file_path = Path(sig_file)
+        if sig_file_path.exists():
+            sig_file_path.unlink()
+        get_gpg_key(sig_file,extra_data_file, gpg_user)
+        base64_encoded_extra_data = base64_encode(extra_data_file)
+        rekor_entry = {
+            "kind": "rekord",
+            "apiVersion": "0.0.1",
+            "spec": {
+                "signature": {
+                    "format": "pgp",
+                    "content": base64_encode(sig_file),
+                    "publicKey": {
+                        "content": base64_encode(public_key_path)
+                    }
+                },
+                "data": {
+                    "content": base64_encoded_extra_data,
+                    "hash": {
+                        "algorithm": "sha256",
+                        "value": hash
+                    }
+                },
+                "extraData": base64_encoded_extra_data
+            }
+        }
+        return rekor_entry;
+
+    def upload_to_rekor(self, rekor_server, extra_data_file, gpg_key, gpg_user):
+        """Runs the step implemented by this StepImplementer.
+
+        Returns
+        -------
+        StepResult
+            Object containing the dictionary results of this step.
+        """
+        rekor_entry = self.create_rekor_entry(gpg_key, gpg_user, extra_data_file)
```

Comment:
i feel like if this is the "upload to rekor" funciton then it should have a parameter for the `rekor_entry` it shouldn't be creating itself. so move this call out of this funciton into `_run_step` and then pass in `rekor_entry` as a paramter to this function. that will simply some of the mocking you have to do in the tests too.